src/Service/DomainFactory.php

```diff
@@ -37,6 +37,9 @@ public function getDomains(string $addressList, Site $site, DetectionResult $det
                 continue;
             }
 
+            // Remove ' and " from beginning and end of domain string if present along with the default "blanks"
+            $domainString = \trim($domainString, " \n\r\t\v\x00'\"");
+
```

Comment:
Rather than removing disallow characters it might be easier (to understand) to allow only some characters, e.g. something like

```suggestion
            $domainString = preg_replace('/[^a-z0-9._-]/', '', $domainString);
```

Bonus question: Where does the extra whitespace and quotation marks come from? Don't the harvesters report clean data?

Reply:
Understand your point. My first go to was to use the predefined `FILTER_SANITIZE_URL` [filter](https://www.php.net/manual/en/filter.filters.sanitize.php) because it would convey the meaning. However ' and " are actually legal characters in an URL, so they don't get stripped. The filter will:

> Remove all characters except letters, digits and $-_.+!*'(),{}|\\^~[]`<>#%";/?:@&=. 

Your suggestion won't work because it will remove all ' and ". However we only want them stripped from beginning and end.

Reply:
Bonus answer: No - by design the harvester i dumb. It dumps data and POST it back, then parsing, validation, processing, normalizing etc is the responsibility of ITKsites.

So this value `COMPOSE_SERVER_DOMAIN='publiccode.foo.bar.dk'` will not have have the `'` stripped by the harvester.